test/react/render-props/react-root-context.js

```diff
@@ -0,0 +1,47 @@
+var React = require('React');
+// the JSX transform converts to React, so we need to add it back in
+this['React'] = React;
+
+var { Provider, Consumer } = React.createContext(null);
+
+function Child(props) {
+  var x = function(context) {
+    var click = function () {
+      return x;
+    }
+
+    return <span onClick={click}>{props.x}</span>
+  }
+
+  return (
+    <div>
+      <Consumer>
+        {x}
+      </Consumer>
+    </div>
+  )
+}
+
+function App(props) {
+  return (
+    <div>
+      <Provider>
```

Comment:
Why no value?

Reply:
It's testing without a value.

Reply:
In that case React will give you `undefined`. But default context value is `null` in this example. Because we render it (and React treats them both as empty), I'm not sure the test would actually catch anything.

Let's change the default value to something like `"foo"`?